core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -230,6 +233,7 @@ public DefaultCacheManager(GlobalConfiguration globalConfiguration, Configuratio
       this.globalComponentRegistry.registerComponent(cacheDependencyGraph, CACHE_DEPENDENCY_GRAPH, false);
       this.globalComponentRegistry.registerComponent(authzHelper, AuthorizationHelper.class);
       this.stats = new CacheContainerStatsImpl(this);
+      this.health = HealthFactory.create(this);
```

Comment:
Currently we have only one Health statistics implementation which is based on JMX. However we might add some more in the future (e.g. Kubernetes might use Prometheus). Implementing a trivial factory for this shows where to plug those. 

If you don't like, I can remove it. But I strongly believe it makes things easier to extend in the future.